controllers/kyma_controller.go

```diff
@@ -418,9 +419,19 @@ func (r *KymaReconciler) handleDeletingState(ctx context.Context, kyma *v1beta2.
 	return false, nil
 }
 
+func (r *KymaReconciler) removeAllFinalizersAndUpdateKyma(ctx context.Context, kyma *v1beta2.Kyma) error {
+	for _, finalizer := range kyma.Finalizers {
+		controllerutil.RemoveFinalizer(kyma, finalizer)
+	}
+	return r.updateKyma(ctx, kyma)
```

Comment:
If you don't mind, it will be nice to separate the logic between removing `Finalizer` and update the kyma CR. What do you think?